modules/product/src/main/java/com/opengamma/strata/product/GenericSecurityPosition.java

```diff
@@ -0,0 +1,663 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.DerivedProperty;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableDefaults;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * A position in a security, where the security is embedded ready for mark-to-market pricing.
+ * <p>
+ * This represents a position in a security, defined by long and short quantity.
+ * The security is embedded directly, however the underlying product model is not available.
+ * The security may be of any kind, including equities, bonds and exchange traded derivatives (ETD).
+ * <p>
+ * The net quantity of the position is stored using two fields - {@code longQuantity} and {@code shortQuantity}.
+ * Each field must be positive, however they may both be non-zero.
```

Comment:
To me "positive" implies non-zero. "These fields must not be negative" sounds better.